cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.service;
+
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.lib.IndexedTable;
+import co.cask.cdap.common.ServiceUnavailableException;
+import co.cask.cdap.data2.datafabric.dataset.DatasetsUtil;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.DatasetManagementException;
+import co.cask.cdap.data2.dataset2.tx.Transactional;
+import co.cask.cdap.data2.metadata.dataset.BusinessMetadataDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.tephra.TransactionExecutor;
+import co.cask.tephra.TransactionExecutorFactory;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.base.Supplier;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Iterators;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Implements operations on {@link BusinessMetadataDataset} transactionally.
+ */
+public class BusinessMetadataStore {
+
+  private static final Id.DatasetInstance BUSINESS_METADATA_INSTANCE_ID =
+    Id.DatasetInstance.from(Id.Namespace.SYSTEM, "business.metadata");
+  private final Transactional<BusinessMdsIterable, BusinessMetadataDataset> txnl;
+
+  @Inject
+  public BusinessMetadataStore(TransactionExecutorFactory txExecutorFactory, final DatasetFramework dsFramework) {
+    this.txnl = Transactional.of(txExecutorFactory, new Supplier<BusinessMdsIterable>() {
```

Comment:
We are trying to get rid of `Transactional` as it's confusing and also doesn't provide much convenient in most cases. Try to use `TransactionExecutorFactory` instead. See

https://github.com/caskdata/cdap/blob/develop/cdap-data-fabric/src/main/java/co/cask/cdap/data/view/MDSViewStore.java